github.py

```diff
@@ -0,0 +1,271 @@
+                self.settings["validated"] = True
+            else:
+                self.settings["validated"] = False
+        # always save
+        dataIO.save_json(self.settingPath, self.settings)
+
+    async def _create_digest(self, server):
+        """Grabs the most recent happenings for each repo, prints nicely."""
+        # all updates since "last update"
+        # subtracting these datetime objects should give current time "minus" the interval
+        # that is, the time interval seconds ago
+        seconds = self.settings["servers"][server.id]["interval"]
+        d = datetime.utcnow() - timedelta(seconds=seconds)
+        # GitHub API takes ISO 8601 format for dates/times
+        # so we need just a little bit more formatting
+        datestring = d.isoformat()[:-7] + "Z"
+        # setting the 'since' parameter in a request for issues
+        # only retrieves issues after a certain time
+        parameters = {"since": datestring}
+        for repo in self.settings["servers"][server.id]["repos"]:
+            print("Checking repo {}...".format(repo))
+            site = "https://api.github.com/repos/{}/issues".format(repo)
+            async with aiohttp.get(site.format(repo), params=parameters) as response:
+                status = response.status
+                reason = response.reason
+                data = await response.json()
+            # display response code and reason, for debugging
+            # ret = "```{}: {}```".format(status, reason)
+            # await self.bot.send_message(ret, channel)
+            if status == 200:
+                if self.settings["servers"][server.id]["notification_method"][0] == "message":
+                    channelId = self.settings["servers"][server.id]["notification_method"][1]
+                    dest = self.bot.get_channel(channelId)
+                    for issue in data:
+                        await self.bot.send_message(self._format_issue(issue, repo), dest)
+                elif self.settings["notification_method"][0] == "webhook":
+                    await self._fire_hooks(server)
+
+    async def _wait_for_updates(self):
+        """Checks for updates from assigned GitHub repos at given interval."""
+        await self.bot.wait_until_ready()
+        # loops until bot stops functioning
+        referenceCount = 0
+        while not self.bot.is_closed:
+            for server in self.bot.servers:
+                # checks if current second count since startup is evenly
+                # divisible by each server's set interval, effectively firing
+                # digests at each server's interval
+                intrval = self.settings["servers"][server.id]["interval"]
+                if referenceCount % intrval == 0:
+                    # notification when scheduled digests fire
+                    print("Scheduled digests firing!")
+                    await self._create_digest(server)
+            await asyncio.sleep(1)
+            referenceCount += 1
```

Comment:
this is not an accurate clock. if you're looking for more accuracy, please see how other cogs do it
here's an example in [26's trigger](https://github.com/Twentysix26/26-Cogs/blob/master/trigger/trigger.py#L502)